src/components/Select/Select.js

```diff
@@ -23,37 +24,41 @@ export const Select = ({
   formTouched,
   labelCopy,
   name,
+  initialValue,
+  options,
   ...rest
 }) => {
-  const onChangeHandler = (event) => {
-    updateSelectedValue(event.value)
-
-    if (onChange) {
-      onChange(event)
-    }
-  }
-
   const resolvedValidator = validator ? validator : () => ''
+  const [selectedOption, updateSelectedOption] = useState(initialValue)
+  const [touched, setTouched] = useState(false)
   const [getError, setError, , validate] = useErrorMessage(resolvedValidator)
-  const [selectedValue, updateSelectedValue] = useState(undefined)
+  const [doValidation] = useInputValidation({
+    validate,
+    setError,
+    formChangeHandler,
+  })
 
-  const validationSelect = () => {
-    const errorMessage = validate(selectedValue)
-    setError(errorMessage)
-    if (formChangeHandler) {
-      formChangeHandler(selectedValue, errorMessage)
-    }
-  }
+  const getSelectedValue = () =>
+    selectedOption ? selectedOption.value : undefined
 
   useEffect(() => {
-    const isSelectedValue = typeof selectedValue !== 'undefined'
-    if (isSelectedValue) {
-      validationSelect()
+    if (!!formChangeHandler && touched) {
+      doValidation(getSelectedValue(), touched)
+    }
+  }, [selectedOption])
+
+  const onChangeHandler = (event) => {
+    updateSelectedOption(options.find((option) => option.value === event.value))
+    setTouched(true)
+
+    if (onChange) {
+      onChange(event)
     }
-  }, [selectedValue])
+  }
 
   const onBlur = () => {
-    validationSelect()
+    setTouched(true)
+    doValidation(getSelectedValue(), true)
```

Comment:
@roblevintennis I noticed that in CheckboxInput we don't check if formChangeHandler exists before validating in onBlur like we do in useEffect. I copied that pattern here, but I don't know why we wouldn't need that here as well.

Reply:
Yeah, I think the `doValidation` is coming from the `useInputValidation` hook which is doing it for us perhaps